pyScienceMode2/sciencemode.py

```diff
@@ -45,38 +46,61 @@ class RehastimGeneric:
         Stuffed byte of protocol.
     """
 
-    VERSION = 0x01
-
-    START_BYTE = 0xF0
-    STOP_BYTE = 0x0F
-    STUFFING_BYTE = 0x81
-    STUFFING_KEY = 0x55
-    MAX_PACKET_BYTES = 69
-    STUFFED_BYTES = [240, 15, 129, 85, 10]
-
-    BAUD_RATE = 460800
-
-    def __init__(self, port: str, show_log: bool = False, with_motomed: bool = False):
+    def __init__(self, port: str, show_log: bool | str = False, with_motomed: bool = False, device_type: str = None):
         """
         Init the class.
 
         Parameters
         ----------
         port : str
             COM port of the Rehastim.
-        show_log : bool
-            Tell if the log will be displayed (True) or not (False).
+        show_log: bool | str
+            If True, all logs of the communication will be printed.
+            If "Partial", only specific logs will be printed.
+            If False, no logs will be printed.
         with_motomed : bool
             If the motomed is connected to the Rehastim, put this flag to True.
-        """
-        self.port = serial.Serial(
-            port,
-            self.BAUD_RATE,
-            bytesize=serial.EIGHTBITS,
-            parity=serial.PARITY_EVEN,
-            stopbits=serial.STOPBITS_ONE,
-            timeout=0.1,
-        )
+        device_type : str
+            Device type. Can be either "Rehastim2" or "RehastimP24".
+        """
+        self.device_type = device_type
+        self.port_name = port
+        if self.device_type == "Rehastim2":
+            self.BAUD_RATE = 460800
```

Comment:
Why do you need it as self argument? is it called after ? Also respect the lower case.

Reply:
May be the arguments only used in the open serial can be remove from the self.

Reply:
Done.